stdlib/timezone/timezone.go

```diff
@@ -0,0 +1,51 @@
+package timezone
+
+import (
+	"context"
+
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/internal/zoneinfo"
+	"github.com/influxdata/flux/interpreter"
+	"github.com/influxdata/flux/runtime"
+	"github.com/influxdata/flux/values"
+)
+
+const pkgpath = "timezone"
+
+func Location(args interpreter.Arguments) (values.Value, error) {
+	name, err := args.GetRequiredString("name")
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := zoneinfo.LoadLocation(name); err != nil {
+		return nil, errors.Wrap(err, codes.Invalid)
+	}
+	return values.BuildObjectWithSize(2, func(set values.ObjectSetter) error {
+		set("zone", values.NewString(name))
+		set("offset", values.NewDuration(values.Duration{}))
+		return nil
+	})
+}
+
+// function is a function definition.
+type function func(args interpreter.Arguments) (values.Value, error)
+
+// makeFunction will construct a values.Function from a function definition.
+func makeFunction(name string, fn function) values.Function {
+	mt := runtime.MustLookupBuiltinType(pkgpath, name)
+	return values.NewFunction(name, mt, func(ctx context.Context, args values.Object) (values.Value, error) {
+		return interpreter.DoFunctionCall(fn, args)
+	}, false)
+}
+
+// registerFunction will create a function from the function definition
+// and register it in this package with the given name.
+func registerFunction(name string, fn function) {
+	runtime.RegisterPackageValue(pkgpath, name, makeFunction(name, fn))
```

Comment:
Why do we use `makeFunction` here? Why not just use `values.NewFunction` like this https://github.com/influxdata/flux/blob/a7853f9cc14cfb58c8f18841d5b6e12ac7b94b99/stdlib/json/encode.go#L16 ? 

In other words why do we need the abstraction through interpreter.DoFunctionCall?

Reply:
I just copied it from another package. It helps me to not repeat myself since the same variables are used in different parts. I can get rid of it but I just found it easier to copy. Might be good to abstract it out into a package.

Reply:
Which package did you copy it from. Perhaps we should simplify both locations so its doesn't get repeated.

Reply:
Looks like I copied it from `dict`.